docs/multiple_image_support_design_proposal.md

```diff
@@ -0,0 +1,549 @@
+fragmentation- Feature Name: Multiple Image Support
+- Start Date: 2018-07-06
+- RFC PR:
+
+# Summary
+Firmware for ARMv8-M MCUs are often released in two separate binary images:
+  - secure code
+  - non-secure code
+
+Currently MCUBoot can only handle single firmware image. MCUBoot could be
+extended to handle the secure and non-secure binaries in one single blob and do
+the upgrade of the images in one round.
+This single blob approach has some drawbacks:
+  - It always needs to update both images even if one of them has not changed.
+  - In multi vendor scenario it requires to share the firmware images among the
+    parties. Because the concatenated firmware must be signed by that vendor,
+    who has access to the private key.
```

Comment:
This would require different parts of the image to have separate signatures, not necessarily that the upgrade happen independently.

Reply:
That's right. But then somehow must be described in header or TLV section that which part of the image is covered by a given signature.

Reply:
If the memory is not contiguous then this probably not feasible.